firedrake/adjoint/blocks.py

```diff
@@ -525,8 +525,13 @@ def evaluate_adj_component(self, inputs, adj_inputs, block_variable, idx, prepar
         """
         if len(adj_inputs) > 1:
             raise NotImplementedError("Interpolate block must have a single output")
-        dJdm = self.backend.derivative(prepared, inputs[idx])
-        return self.backend.Interpolator(dJdm, self.V).interpolate(adj_inputs[0], transpose=True).vector()
+        input = inputs[idx]
+        dJdm = self.backend.derivative(prepared, input)
+        # Get the function space from `dJdm` argument
+        arg, = extract_arguments(dJdm)
+        # Make sure to have a cofunction output
+        output = self.backend.Cofunction(arg.function_space().dual())
+        return self.backend.Interpolator(dJdm, self.V).interpolate(adj_inputs[0], output=output, transpose=True).vector()
```

Comment:
Should this still return a `vector`? Is that what's required for compatibility with the innards of pyadjoint?

Reply:
Yes it is.